### PR TITLE
feat: t4 - get filename without the extension

### PR DESCRIPTION
**Pull request by KrawczowaKris**
# Summary
## Test Task
**T4 - Get filename without the extension**
Script should get a folder as an argument and return all files (including in the nested directories) but without the extension.

## my results
`./chapter2-l2-t4-files-no-extensions.sh /home/nadya/fig`

**output:**
```
/home/nadya/fig/e
/home/nadya/fig/d
/home/nadya/fig/wer
/home/nadya/fig/qw/asd
/home/nadya/fig/qw/uop
/home/nadya/fig/fds/dfs
/home/nadya/fig/fds/rert/d
/home/nadya/fig/fds/rert/d
/home/nadya/fig/fds/rert/poi
/home/nadya/fig/sss/reter
/home/nadya/fig/sss/wertep
```

### CODE REVIEW

File: chapter2/tests/t4/chapter2-l2-t4-files-no-extensions.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+# Script should get a folder as an argument and return all files
+# (including in the nested directories) but without the extension
+
+PATH_TO_FOLDER=$1
+
+files=$(find $PATH_TO_FOLDER -type f)
+
+for file in $files; do
```
Comment (kseniyashaydurova) [inline chapter2/tests/t4/chapter2-l2-t4-files-no-extensions.sh:10]: Not working correctly when there is a `.test` folder inside main folder:
image

I suppose that there should be output:
```
test/.test/
test/.test/1
test/.test/2
test/
test/1
test/2
```

**Review by kseniyashaydurova — CHANGES_REQUESTED** [review]

File: chapter2/tests/t4/chapter2-l2-t4-files-no-extensions.sh (hunk 1 above)
Comment (KrawczowaKris) [reply chapter2/tests/t4/chapter2-l2-t4-files-no-extensions.sh:10]: done

File: chapter2/tests/t4/chapter2-l2-t4-files-no-extensions.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
```
Comment (kseniyashaydurova) [inline chapter2/tests/t4/chapter2-l2-t4-files-no-extensions.sh:1]: Script works incorrectly still

image

1. It incorrectly processes double extensions (i.e. in real life there can be `archive.tar.gz` files with double extensions)
2. It incorrectly processes spaces in files and folders names

File: chapter2/tests/t4/chapter2-l2-t4-files-no-extensions.sh [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
```
Comment (kseniyashaydurova) [inline chapter2/tests/t4/chapter2-l2-t4-files-no-extensions.sh:2]: Also I checked `node_modules` dir for curiosity and it also doesn't work correctly
image

File: chapter2/tests/t4/chapter2-l2-t4-files-no-extensions.sh [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+# Script should get a folder as an argument and return all files
+# (including in the nested directories) but without the extension
+
+PATH_TO_FOLDER="$1"
+
+files=$(find "${PATH_TO_FOLDER}" -type f)
+
+for file in $files; do
+  if [[ "${file}" == *"/."* && \
```
Comment (kseniyashaydurova) [inline chapter2/tests/t4/chapter2-l2-t4-files-no-extensions.sh:11]: Please separate this `magic` if condition to separate vars and name them properly, so we could see what kind of magic happens here, now it is too cryptic and needs comments too

**Review by kseniyashaydurova — CHANGES_REQUESTED** [review]

File: chapter2/tests/t4/chapter2-l2-t4-files-no-extensions.sh (hunk 4 above)
Comment (KrawczowaKris) [reply chapter2/tests/t4/chapter2-l2-t4-files-no-extensions.sh:11]: done

File: chapter2/tests/t4/chapter2-l2-t4-files-no-extensions.sh (hunk 2 above)
Comment (KrawczowaKris) [reply chapter2/tests/t4/chapter2-l2-t4-files-no-extensions.sh:1]: done

File: chapter2/tests/t4/chapter2-l2-t4-files-no-extensions.sh [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+  # Check if the first character of the filename is a dot
+  if [[ "${filename:0:1}" == "." ]]; then
+    # Compare the number of dots in the file name with one
+    if [[ $(echo "${filename}" | tr -cd '\.' | wc -c) -eq 1 ]]; then
+      echo "${filefolder}/${filename}"
+    else
+      file_output "${filename}" "y"
+    fi
+    continue
```
Comment (kseniyashaydurova) [inline chapter2/tests/t4/chapter2-l2-t4-files-no-extensions.sh:41]: I think continue is redundant here

**Review by kseniyashaydurova — APPROVED** [review]

**Pull request merged**